provisioner/ansible-local/provisioner.go

```diff
@@ -549,8 +549,12 @@ func (p *Provisioner) invokeGalaxyCommand(args []string, ui packersdk.Ui, comm p
 func (p *Provisioner) executeAnsible(ui packersdk.Ui, comm packersdk.Communicator) error {
 	inventory := filepath.ToSlash(filepath.Join(p.config.StagingDir, filepath.Base(p.config.InventoryFile)))
 
+	packerHTTPAddr := p.generatedData["PackerHTTPAddr"]
+	if packerHTTPAddr == nil {
+		packerHTTPAddr = ""
```

Comment:
I would suggest maybe not adding the value to the extra-vars instead?
Unless this is used in the target, but then I would assume that it being empty would probably yield an error, so it may or may not be a good idea, what do you think?

Reply:
Thanks for your reply and suggestion. Your idea is better. The ansible-playbook already provides a good handler for empty/default extra-vars, so I believe there will be no potential error. It's unnecessary to add them explicitly when they are empty.

I'm testing new codes with my module. Then I'll update this patch once I get any updates.